### PR TITLE
Ft/back end

### LINKED ISSUE #2: /paraphrase endpoint to paraphrase a single sentence
As a User of ParaphrAce, I want to be able to paraphrase a sentence, so that I can use it in my paper without plagiarising.

- [x] Implement model
- [x] implement route /paraphrase
- [x] integrate storage provider
- [x] integrate db
- [x] unit tests

### LINKED ISSUE #3: /user endpoint for creating users
Create /user endpoint for creating users

### LINKED ISSUE #4: backend server with rest endpoints
create backend server with rest endpoints

### PR DESCRIPTION
**Pull request by michmich112**
Implementation for back end server with routes:
- `/api/user/create`: Create a new user and retrieve a session token
- `/api/paraphrase/create`: Create a new paraphrase

Updated return of model to include start & end times of the paraphrase generation

Dockerized the server and created a docker-compose configuration for local testing.

Closes #2 
Closes #3 
Closes #4

### CODE REVIEW

File: server/README.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+Exposes the api on port 8080 by default.
+
+## Requirements
+You will need the following to run the server:
+- PostgreSQL Database: A postgresql database (v9 or higher);
+- AWS S3 Bucket: An S3 bucket with credentials for an IAM account with the permissions to put objects;
+- Paraphrasing Api: A Paraphrasing API, specifically the one in this monorepo that implements the Pegasus model.
+
+> Note: The PostgreSQL database must be using version 9 or later
+
```
Comment (languantan) [inline server/README.md:12]: hmm, how do you run the tests?

File: server/middleware/create-user_test.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+	if status := rr.Code; status != http.StatusCreated {
+		t.Errorf("Expected status code Created, received %v", status)
+	}
+
+	user := models.User{}
+	err = json.NewDecoder(rr.Body).Decode(&user)
+
+	if err != nil {
+		t.Fatal(err)
```
Comment (languantan) [inline server/middleware/create-user_test.go:53]: okay understand. just used to frameworks with expect/should syntax

**Review by languantan — APPROVED** [review]

File: server/README.md (hunk 1 above)
Comment (michmich112) [reply server/README.md:12]: Added documentation in latest commit

File: server/middleware/create-user_test.go (hunk 2 above)
Comment (michmich112) [reply server/middleware/create-user_test.go:53]: Undestood, this is the testing suite that is built into the language

**Pull request merged**